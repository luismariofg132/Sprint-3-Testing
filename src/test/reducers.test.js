import { loginReducer, typeUserReducer } from "../redux/reducers/reducers"
import { types } from "../redux/types/types"
import { shallow } from 'enzyme'
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'
import PrivateRouter from "../routers/PrivateRouter"
import { Dasboard } from "../routers/Dashboard"
import AdminRouters from "../routers/AdminRouters"
import Crud from "../components/control/Crud"

describe('Pruebas de usuarios', () => {
    // Prueba de que al iniciar sesion el reducer nos devuelva el usuario
    test('Prueba de usuario registrado', () => {
        const action = {
            type: types.login,
            payload: {
                id: "XzghYqMSGgZrQC1TaVf5w39oKLn1",
                displayname: "kekene1002@sueshaw.com"
            }
        }

        const initialState = {}

        const stado = loginReducer(initialState, action)
        expect(stado).toEqual({
            id: "XzghYqMSGgZrQC1TaVf5w39oKLn1",
            name: "kekene1002@sueshaw.com"
        })
    })

    // Prueba que el usuario que se le envia no reciba permisos de administrados
    test('El usuario admin sea igual false', () => {
        const action = {
            type: types.typeUser,
            payload: false
        }
        const initialState = false
        const userAdmin = typeUserReducer(initialState, action)
        expect(userAdmin).toEqual({ "admin": false })
    })

    test('Muestre cuando el usuario esta autenticado', () => {
        const isAuthenticated = false

        const wrapper = shallow(
            <Router>
                <Routes>
                    <Route path='/*' element={<PrivateRouter isAuthenticated={isAuthenticated}>
                        <Dasboard />
                    </PrivateRouter>} />
                </Routes>
            </Router>
        )
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('Router')).toEqual(isAuthenticated)
    })

    test('Muestra cuando el usuario es administrador', () => {
        const isAdmin = true
        const wrapper = shallow(
            <Router>
                <Routes>
                    <Route path='/controlDatos' element={<AdminRouters isAdmin={isAdmin}>
                        <Crud />
                    </AdminRouters>} />
                </Routes>
            </Router>
        )
        expect(wrapper).toMatchSnapshot()
    })
})