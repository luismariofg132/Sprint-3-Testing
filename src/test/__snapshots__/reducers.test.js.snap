// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de usuarios Muestra cuando el usuario es administrador 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes>
    <Route
      element={
        <AdminRouters
          isAdmin={true}
        >
          <Crud />
        </AdminRouters>
      }
      path="/controlDatos"
    />
  </Routes>
</Router>
`;

exports[`Pruebas de usuarios Muestre cuando el usuario esta autenticado 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes>
    <Route
      element={
        <PrivateRouter
          isAuthenticated={false}
        >
          <Dasboard />
        </PrivateRouter>
      }
      path="/*"
    />
  </Routes>
</Router>
`;
